types/cloudflare/index.d.ts

```diff
@@ -54,34 +54,72 @@ declare namespace Cloudflare {
     interface SrvDnsRecord {
         type: 'SRV';
         data: {
-          name: string;
-          service: string;
-          proto: string;
-          ttl: number;
-          proxied?: boolean | undefined;
-          priority: number;
-          weight: number;
-          port: number;
-          target: string;
+            name: string;
+            service: string;
+            proto: string;
+            ttl: number;
+            proxied?: boolean | undefined;
+            priority: number;
+            weight: number;
+            port: number;
+            target: string;
         };
     }
 
     type DnsRecord = DnsRecordWithPriority | DnsRecordWithoutPriority | SrvDnsRecord;
+    type DnsRecordByType<RecordType extends RecordTypes> = RecordType extends 'MX' | 'URI'
+        ? DnsRecordWithPriority
+        : RecordType extends 'SRV'
+        ? SrvDnsRecord
+        : RecordType extends Exclude<RecordTypes, 'MX' | 'SRV' | 'URI'>
+        ? DnsRecordWithoutPriority
+        : DnsRecord;
 
     interface DNSRecords {
-        edit(
+        edit(zone_id: string, id: string, record: DnsRecord): ResponseObjectPromise;
+        browse<RecordType extends RecordTypes = any>(
             zone_id: string,
-            id: string,
-            record: DnsRecord,
-        ): ResponseObjectPromise;
-        browse(zone_id: string): ResponseObjectPromise;
+            options?: DnsRecordsBrowseOptions<RecordType>,
+        ): Promise<DnsRecordsBrowseResponse<RecordType>>;
         export(zone_id: string): ResponseObjectPromise;
         del(zone_id: string, id: string): ResponseObjectPromise;
         read(zone_id: string, id: string): ResponseObjectPromise;
-        add(
-            zone_id: string,
-            record: DnsRecord,
-        ): ResponseObjectPromise;
+        add(zone_id: string, record: DnsRecord): ResponseObjectPromise;
+    }
+
+    interface DnsRecordsBrowseOptions<RecordType extends RecordTypes> {
+        page?: number;
+        per_page?: number;
+        name?: string;
+        content?: string;
+        type?: RecordType;
+        order?: 'type' | 'name' | 'content' | 'ttl' | 'proxied';
+        direction?: 'asc' | 'desc';
+        match?: 'any' | 'all';
+        tag?: string;
+        tag_match?: 'any' | 'all';
+        search?: string;
+        comment?: string;
+        // TODO: support nested filters (for example tag.absent)
+    }
+
+    interface DnsRecordsBrowseResponse<RecordType extends RecordTypes> {
+        result: Array<DnsRecordByType<RecordType>> | null;
+        result_info: {
+            page: number;
+            per_page: number;
+            total_pages: number;
```

Comment:
This property doesn't exist on the docs.

Reply:
Great catch, my bad 🙏